components/tools/OmeroWeb/test/integration/test_api_wells.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (C) 2017 University of Dundee & Open Microscopy Environment.
+# All rights reserved.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests querying Wells with web json api."""
+
+from omeroweb.testlib import IWebTest, _get_response_json
+from django.core.urlresolvers import reverse
+from django.conf import settings
+import pytest
+from omero.gateway import BlitzGateway
+from omero_marshal import get_encoder
+from omero.model import PlateI, WellI, WellSampleI, ImageI, LengthI
+from omero.model.enums import UnitsLength
+from omero.rtypes import rstring, rint, unwrap
+import json
+
+
+def get_update_service(user):
+    """Get the update_service for the given user's client."""
+    return user[0].getSession().getUpdateService()
+
+
+def get_query_service(user):
+    """Get the query_service for the given user's client."""
+    return user[0].getSession().getQueryService()
+
+
+def get_connection(user, group_id=None):
+    """Get a BlitzGateway connection for the given user's client."""
+    connection = BlitzGateway(client_obj=user[0])
+    # Refresh the session context
+    connection.getEventContext()
+    if group_id is not None:
+        connection.SERVICE_OPTS.setOmeroGroup(group_id)
+    return connection
+
+
+def cmp_name_insensitive(x, y):
```

Comment:
probably time to have helper methods since this one for example is now in all the test files